pkg/plugin/run.go

```diff
@@ -175,17 +177,33 @@ func (e *AutoscaleEnforcer) handleAgentRequest(
 	resp := api.PluginResponse{
 		Permit:      permit,
 		Migrate:     migrateDecision,
-		ComputeUnit: *node.computeUnit,
+		ComputeUnit: getComputeUnitForResponse(node, supportsFractional),
 	}
 	pod.mostRecentComputeUnit = node.computeUnit // refer to common allocation
 	return &resp, 200, nil
 }
 
+// getComputeUnitForResponse tries to return compute unit that the agent supports
+// if we have a fractional CPU but the agent does not support it we multiply the result
+func getComputeUnitForResponse(node *nodeState, supportsFractional bool) api.Resources {
+	computeUnit := *node.computeUnit
+	if !supportsFractional {
+		initialCU := computeUnit
+		i := uint16(2)
+		for ; computeUnit.VCPU < 1000; i++ {
+			computeUnit = initialCU.Mul(i)
+		}
+	}
+
+	return computeUnit
+}
+
 func (e *AutoscaleEnforcer) handleResources(
 	pod *podState,
 	node *nodeState,
 	req api.Resources,
 	startingMigration bool,
+	suportsFractional bool,
```

Comment:
Also: should have some validation here, returning HTTP 400 if `!supportsFractionalCPU && req.VCPU % 1000 != 0`